File: packages/shared-types/src/analytics.ts
// Types for anonymous analytics events and the dashboard summary.
export type AnalyticsEvent = {
  event_id: string;
  event_type: string;
  session_id: string;
  timestamp: string;
  metadata: Record<string, string | number | boolean>;
};

export type DashboardSummary = {
  total_sessions: number;
  total_routes_requested: number;
  total_steps_completed: number;
  total_fallbacks: number;
  top_locations: Array<{ location_id: string; count: number }>;
  recent_fallbacks: Array<{ event_id: string; reason: string; timestamp: string }>;
};

File: services/navigation-engine/app/models/analytics.py
"""Analytics event and dashboard models."""

from __future__ import annotations

from typing import Any, Optional

from pydantic import BaseModel, Field


class AnalyticsEventRequest(BaseModel):
    event_type: str
    session_id: Optional[str] = None
    metadata: dict[str, Any] = Field(default_factory=dict)


class AnalyticsEvent(BaseModel):
    event_id: str
    event_type: str
    session_id: Optional[str] = None
    timestamp: str
    metadata: dict[str, Any] = Field(default_factory=dict)


class TopLocation(BaseModel):
    location_id: str
    count: int


class RecentFallback(BaseModel):
    event_id: str
    reason: str
    timestamp: str


class DashboardSummary(BaseModel):
    total_sessions: int
    total_routes_requested: int
    total_steps_completed: int
    total_fallbacks: int
    top_locations: list[TopLocation] = Field(default_factory=list)
    recent_fallbacks: list[RecentFallback] = Field(default_factory=list)
